### PR TITLE
Add chapter about the published container registry 

### PR DESCRIPTION
**Pull request by dcermak**
The user guide does not mention the container registry at all, especially that you can use the published gpg signatures with podman for verification. This PR adds a rudimentary guide how to use this feature.

cc @tomschr @adrianschroeter

### CODE REVIEW

File: xml/obs_build_containers.xml [hunk 1]
```diff
@@ -14,34 +14,36 @@
    <remark>TBD</remark>
   </abstract>
  </info>
- <para> Containers are workloads which container all necessary files to make
+ <para> Containers are workloads which embed all necessary files to make
   the workload independent of the running host OS. This includes (but is not
   limited to) libraries, executables and shared resource files. </para>
 
  <sect1>
```
Comment (sknorr) [inline xml/obs_build_containers.xml:21]: I'd suggest adding a `xml:id=...` attribute to all sections, for easier referencing from other parts of the book (and for use as permalinks to a particular section.

File: xml/obs_build_containers.xml [hunk 2]
```diff
@@ -91,4 +93,170 @@
...
+           <literal>latest</literal> with alternatives being provided in a similar
+           fashion as the image's name.
+         </para>
+       </listitem>
+     </itemizedlist>
+   </para>
+   <para>
+     The <link
+     xlink:href="https://github.com/openSUSE/cooverview">cooverview</link>
+     project provides a simple user facing webpage to search for containers
```
Comment (sknorr) [inline xml/obs_build_containers.xml:156]: Grammar:
```suggestion
     project provides a simple user-facing webpage to search for containers
```

File: xml/obs_build_containers.xml [hunk 3]
```diff
@@ -91,4 +93,170 @@
...
+   </para>
+   <para>
+     Podman has to be configured first as outlined in the following steps.
+     <itemizedlist>
+       <listitem>
+         <para>
+           Create a <literal>yaml</literal> file under
+           <literal>/etc/containers/registries.d/</literal> with an appropriate
+           name for your instance and the following contents:
+           <example>
```
Comment (sknorr) [inline xml/obs_build_containers.xml:182]: For layout improvements/simplicity, avoid nesting screens/examples/tables/figures/lists/... within `<para/>`. Instead close the para and start a new para later (i.e. on line 191):
```suggestion
         </para>
         <example>
```

This appears a few times below as well.

File: xml/obs_build_containers.xml [hunk 4]
```diff
@@ -91,4 +93,170 @@
...
+             <title>registry.yaml</title>
+             <screen>
+---
+docker:
+  registry.opensuse.org:
+    sigstore: https://registry.opensuse.org/sigstore
+             </screen>
+           </example>
+           Where you have to replace <literal>registry.opensuse.org</literal>
```
Comment (sknorr) [inline xml/obs_build_containers.xml:191]: 1. Shorten:

```suggestion
           Replace <literal>registry.opensuse.org</literal>
```

2. DocBook knows the element `<replaceable>` (shown in italics in our documents) which can be used like this:

```
  <replaceable>REGISTRY_URL</replaceable>:
    sigstore: https://<replaceable>REGISTRY_URL</replaceable>/sigstore

[...]

    Replace <replaceable>REGISTRY_URL</replaceable> with the appropriate URL to your
    instance of &obs; (for example, <literal>registry.opensuse.org</literal>.
```

File: xml/obs_build_containers.xml [hunk 5]
```diff
@@ -91,4 +93,170 @@
...
+           with the appropriate URL to your instance of the Open Build Service.
+         </para>
+       </listitem>
+       <listitem>
+         <para>
+           Add the following object into the key <literal>transports</literal>
+           in the file <literal>/etc/containers/policy.json</literal>:
```
Comment (sknorr) [inline xml/obs_build_containers.xml:198]: Ideally, use filename:
```suggestion
           in the file <filename>/etc/containers/policy.json</filename>:
```

File: xml/obs_build_containers.xml [hunk 6]
```diff
@@ -91,4 +93,170 @@
...
+        {
+          "type": "signedBy",
+          "keyType": "GPGKeys",
+          "keyPath": "/path/to/my/gpg_public.key"
+        }
+      ]
+    }
+             </screen>
+           </example>
+           The complete <literal>/etc/containers/policy.json</literal> can then
```
Comment (sknorr) [inline xml/obs_build_containers.xml:213]:
```suggestion
           The complete <filename>/etc/containers/policy.json</filename> can then
```

File: xml/obs_build_containers.xml [hunk 7]
```diff
@@ -91,4 +93,170 @@
...
+  }
+}
+             </screen>
+           </example>
+         </para>
+       </listitem>
+       <listitem>
+         <para>
+           Save the public key of the project where your image is build under
+           <literal>/path/to/my/gpg_public.key</literal> (you may use a
```
Comment (sknorr) [inline xml/obs_build_containers.xml:250]:
```suggestion
           <filename>/path/to/my/gpg_public.key</filename> (you may use a
```

File: xml/obs_build_containers.xml [hunk 8]
```diff
@@ -91,4 +93,170 @@
...
+           different location, if you also specify it in
+           <literal>/etc/containers/policy.json</literal>).
```
Comment (sknorr) [inline xml/obs_build_containers.xml:252]:
```suggestion
           <filename>/etc/containers/policy.json</filename>).
```

File: xml/obs_build_containers.xml (hunk 4 above)
Comment (dcermak) [reply xml/obs_build_containers.xml]: Thanks, this is pretty nifty and I've now included it in another place as well!

I have a tiny gripe with the styling though, the italics make the `/` after `REGISTRY_URL` harder to see imho:
image

Comment (sknorr) [reply xml/obs_build_containers.xml]: Right, italics don't always play well with punctuation and in this case make it a bit harder to discern whether the following character is a slash `/` or a pipe `|`. Not sure I have a better idea for formatting right now -- do you?

Comment (dcermak) [reply xml/obs_build_containers.xml]: Not really, no. I'd keep it this way, to be consistent with the rest of the documentation.

File: xml/obs_build_containers.xml [hunk 9]
```diff
@@ -70,8 +72,8 @@
    <varlistentry>
     <term>Flatpak</term>
     <listitem>
-     <para>Flatpak is currently not supported due to the lack of
-    reproducible builds.</para>
+     <para>Flatpak packages can be build in the Open Build Service, see
+     &flatpak; for further details.</para>
```
[the remark this answers is not in the document]
Comment (dcermak) [reply xml/obs_build_containers.xml:75]: I've added a link to the flatpak section.

File: xml/obs_build_containers.xml (hunk 1 above)
Comment (dcermak) [reply xml/obs_build_containers.xml]: I've added those as well.

**Comment by sknorr** [comment]
Thank you -- I took the liberty of merging. (And hope that's fine.)

**Pull request merged**